dev/remsh-tls

```diff
@@ -24,6 +24,6 @@ if [ -z $HOST ]; then
 fi
 
 NAME="remsh$$@$HOST"
```

Comment:
will we need to generate a remsh certificate dynamically now?

Reply:
`erl -name $NAME -remsh "$NODE@$HOST" -hidden -proto_dist inet_tls -ssl_dist_optfile "src/couch_dist/certs/out/${NODE}-dist.conf"`

Maybe not, since we're using a dynamic `${NODE}-dist.conf` and certificates.

Reply:
despite the code in `inet_tls_dist.erl` that looks for the dirname entry (and parses it out in cert_nodes/1), it appears to be ignored.

It suffices to have a subjectAltName of the hostname. My elixir-certs fork is unnecessary and should not be used.

The cert resulting from

```
./certs create-cert --issuer-cert out/ca-cert.pem --issuer-key out/ca-key.pem --out-cert out/node-cert.pem --out-key out/node-key.pem --template server --subject '/CN=127.0.0.1'```

is sufficient and can be used as the client and server cert of all nodes with the same hostname, which is all of the dev nodes.

Obviously a 'real' cluster would have distinct ip's or hostnames for each node, though.

Reply:
So

```
commit 5586959ed008c09141689f1e8865476150e48519
Author: Raimo Niskanen <raimo@erlang.org>
Date:   Thu Apr 26 09:37:24 2018 +0200

    Allow check for node name
```

I think introduced the ability to check the node names also, from the data in the certificate

and

```
commit 794df8cbba8d7942dcb3bf2cbdfa526b04d41dd3
Author: Raimo Niskanen <raimo@erlang.org>
Date:   Fri Jun 8 15:49:06 2018 +0200

    Use public_key to verify client hostname
```
removed it (and used the net_kernel:allow list to restrict node names instead).

they just left the cert_nodes function in but it's unreachable?

Reply:
May I keep your `elixir-certs` fork in this demo, as it might be used for a `real` cluster setup?
I can modify the `gen_cert` script to use the `server` template and point to one conf and certificate.

Reply:
I think it was maybe optional previously with ```{verify_fun, {inet_tls_dist, verify_client, undefined}},``` option, but I am getting confused at this point.

Jessica and I have independently shown that 'node1@127.0.0.1' can start up with node2-dist.conf (which points at node2-cert.pem which has the dirname of 'node2') and still ping 'node2@127.0.0.1' which uses node2-dist.conf also.

So erlang does not check the node name embedded in the cert and apparently cannot be configured to do so.

the `inet_tls_dist:cert_nodes/1` function (and parse_extensions) just exists to tease me.

Reply:
@RaimoNiskanen in case you might find our journey amusing ^.